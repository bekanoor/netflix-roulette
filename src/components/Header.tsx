
function Header(props) {
  return (
    <div>
      wdwd
    </div>
  )
}

export default Header;