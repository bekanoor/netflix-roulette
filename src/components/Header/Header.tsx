import React from 'react'
import Button from '../UiKit/Button'
import Input from '../UiKit/Input'
import P from '../UiKit/typography/P'
import H1 from '../UiKit/typography/H1'
import Ul from '../UiKit/Ul'
import Li from '../UiKit/Li'

const Header = () => {
  return (
    <header className='header'>
      <P
        className='header__logo'
        opacity={1}
        theme='red'
        register='down'
        font='regular'
        weight='regular'
      >
        netflixroulette
      </P>
      <H1>FIND YOUR MOVIE</H1>
      <div>
        <Input placeholder='What do you want to watch?'></Input>
        <Button className='header__search-btn' theme='danger' size='standard' color='white'>SEARCH</Button>
      </div>
      <nav>
        <Ul className='header__nav'>
          <Li>
            <P opacity={1} theme='white' register='up' font='regular' weight='regular'>
              search by
            </P>
          </Li>
          <Li>
            <Button className='header__filter-btn' theme='danger' size='small' color='white'>TITLE</Button>
          </Li>
          <Li>
            <Button className='header__filter-btn' theme='danger' size='small' color='white'>GENRE</Button>
          </Li>
        </Ul>
      </nav>
    </header>
  )
}

export default Header
