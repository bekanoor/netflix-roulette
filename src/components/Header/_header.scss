.header {
  height: 400px;
  background: url('../assets/images/header-bg.png');
  padding: 0 56px;
}