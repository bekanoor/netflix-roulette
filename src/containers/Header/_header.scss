.header {
  height: 400px;
  background: url('../../assets/images/header-bg.png');
  padding: 0 56px;
}
.header__logo{
  margin: 0;
  padding: 20px 0;
}
.header__nav{
  padding: 0;
  width: 300px;
}
.header__search-btn {
  margin-left: 10px;
}
.header__filter-btn {
  margin-top: 15px;
}