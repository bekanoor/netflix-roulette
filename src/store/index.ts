export * from './actions'
export *  from './store'
export * from './moviesReducer'
export * from './searchParamReducer'
