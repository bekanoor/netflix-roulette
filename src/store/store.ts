import { reducer } from './reducer'

import { createStore } from 'redux'


const store = createStore(reducer)

export { store }

export type RootState = ReturnType<typeof store.getState >
export type AppDispatch = typeof store.dispatch  
