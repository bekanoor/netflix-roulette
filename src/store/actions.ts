import { Movie } from 'src/models'

export const setSearchType = (value: 'title' | 'genres') => {
  return { type: 'SET_SEARCH_BY', payload: value }
}

export const setFilterType = (value: string) => {
  return { type: 'SET_FILTER_BY', payload: value }
}

export const setSearchQuery = (value: string) => {
  return { type: 'SET_SEARCH_QUERY', payload: value }
}

export const setSearchButton = (value: boolean) => {
  return { type: 'SET_BUTTON_STATE', payload: value }
}

export const setData = (data: any) => {
  return { type: 'SET_DATA', payload: data }
}

export const setLoading = (status: boolean) => {
  return { type: 'SET_LOADING', payload: status }
}

export const setSelectedMovie = (value: Array<Movie>) => {
  return { type: 'SET_SELECTED_MOVIE_ID', payload: value }
}

export const setSameGenreMovies = (value: Array<Movie>) => {
  return { type: 'SET_SAME_GENRE_MOVIES', payload: value }
}
