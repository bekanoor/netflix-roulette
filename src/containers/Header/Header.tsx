import { setSearchButton, setSearchInput } from '../../store'
import { Button } from '../../components'
import { SearchOption } from '../SearchOption'

import React from 'react'

interface IProps {
  dispatch: (value: object) => void
}

class Header extends React.Component<IProps> {
  handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    this.props.dispatch(setSearchInput(event.target.value.toLocaleLowerCase()))
  }

  buttonHandler = () => {
    this.props.dispatch(setSearchButton('active'))
  }

  render() {
    return (
      <header className='header' data-testid='header-test'>
        <p className='header__logo'>
          netflix
          <span className='header__logo header__logo-regular'>roulette</span>
        </p>
        <h1 className='main-title'>FIND YOUR MOVIE</h1>
        <div>
          <input
            onChange={this.handleInputChange}
            className='input-search header__input'
            placeholder='What do you want to watch?'
          ></input>
          <Button
            className='header__search-btn'
            theme='danger'
            size='standard'
            color='white'
            onClick={this.buttonHandler}
          >
            SEARCH
          </Button>
        </div>
        <div className='header__search-option'>
          <SearchOption></SearchOption>
        </div>
      </header>
    )
  }
}

export { Header }
