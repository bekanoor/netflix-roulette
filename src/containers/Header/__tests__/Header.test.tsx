import { render } from '@testing-library/react'
import { Header } from '../Header'
import '@testing-library/jest-dom'
import { Provider } from 'react-redux'
import { store } from '../../../store'

describe('test header', () => {
  const mockF = (value: object): void => {}

  test('should render header correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <Header dispatch={mockF} />
      </Provider>
    )
    expect(container).toMatchSnapshot()
  })

  test('dom elements really exist', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Header dispatch={mockF} />
      </Provider>
    )

    expect(getByTestId('header-test')).toBeInTheDocument()
  })
})
