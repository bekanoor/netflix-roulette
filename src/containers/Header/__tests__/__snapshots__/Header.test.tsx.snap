// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test header should render header correctly 1`] = `
<div>
  <header
    class="header"
    data-testid="header-test"
  >
    <p
      class="header__logo"
    >
      netflix
      <span
        class="header__logo header__logo-regular"
      >
        roulette
      </span>
    </p>
    <h1
      class="main-title"
    >
      FIND YOUR MOVIE
    </h1>
    <div>
      <input
        class="input-search header__input"
        placeholder="What do you want to watch?"
      />
      <button
        class="sc-bczRLJ borxYo header__search-btn"
        color="white"
      >
        SEARCH
      </button>
    </div>
    <div
      class="header__search-option"
    >
      <nav
        data-testid="search-option-test"
      >
        <ul
          class="option-list"
        >
          <li
            class="option-list__item"
          >
            <p>
              search by
            </p>
          </li>
          <li
            class="option-list__item"
          >
            <button
              class="option-list__button option-list__button--active"
            >
              title
            </button>
          </li>
          <li
            class="option-list__item"
          >
            <button
              class="option-list__button"
            >
              genre
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</div>
`;
